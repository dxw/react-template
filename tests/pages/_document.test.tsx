/* eslint-env jest */
import { advanceTo, clear } from "jest-date-mock";
import React from "react";
import renderer from "react-test-renderer";

import Document from "../../pages/_document";

beforeAll(() => {
  advanceTo(new Date("2019-01-01T00:00:00.000Z"));
});

afterAll(() => {
  clear();
});

it("renders correctly", () => {
  const component = renderer.create(
    <Document
      html="<div>test-html</div>"
      __NEXT_DATA__={{
        dataManager: "test-data-manager",
        props: {},
        page: "/test-page",
        query: {},
        buildId: "test-build"
      }}
      dangerousAsPath="/test-dangerous-as-path"
      ampPath="/test-amp-path"
      inAmpMode={false}
      hybridAmp={false}
      staticMarkup={false}
      isDevelopment={false}
      hasCssMode={false}
      devFiles={[]}
      files={[]}
      dynamicImports={[]}
      canonicalBase="http://localhost:3000/"
    />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
