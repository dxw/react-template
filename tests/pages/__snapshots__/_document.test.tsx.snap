// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<html>
  <head>
    <style>
      
              body {
                font-family: "Lucida Sans", "Lucida Sans Regular",
                  "Lucida Grande", "Lucida Sans Unicode", Geneva, Verdana,
                  sans-serif;
              }
            
    </style>
    <meta
      content="0"
      name="next-head-count"
    />
    <link
      as="script"
      href="undefined/_next/static/test-build/pages/test-page.js?ts=1546300800000"
      rel="preload"
    />
    <link
      as="script"
      href="undefined/_next/static/test-build/pages/_app.js?ts=1546300800000"
      rel="preload"
    />
  </head>
  <body>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<div>test-html</div>",
        }
      }
      id="__next"
    />
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "{\\"dataManager\\":\\"test-data-manager\\",\\"props\\":{},\\"page\\":\\"/test-page\\",\\"query\\":{},\\"buildId\\":\\"test-build\\"}",
        }
      }
      id="__NEXT_DATA__"
      type="application/json"
    />
    <script
      async={true}
      data-next-page="/test-page"
      src="undefined/_next/static/test-build/pages/test-page.js?ts=1546300800000"
    />
    <script
      async={true}
      data-next-page="/_app"
      src="undefined/_next/static/test-build/pages/_app.js?ts=1546300800000"
    />
  </body>
</html>
`;
